### PR TITLE
Adds improved support for testing golang libraries

### PR DESCRIPTION
**Pull request by JosephSalisbury**

### CODE REVIEW

File: workflow/workflow_test.go [hunk 1]
```diff
@@ -111,7 +111,28 @@ func TestGetBuildWorkflow(t *testing.T) {
 			},
 		},
 
-		// Test 3 that a project with a golang file not named `main.go` produces a
+		// Test that a library project creates a correct workflow.
```
Comment (JosephSalisbury) [inline workflow/workflow_test.go]: This is for adding support for testing `microkit` / `operatorkit`, which don't have golang files in their top level directory, so golang workflows were not executed.

File: workflow/workflow.go [hunk 2]
```diff
@@ -99,11 +92,17 @@ func NewBuild(projectInfo ProjectInfo, fs afero.Fs) (Workflow, error) {
...
 		w = append(w, goTest)
 
-		goBuild, err := NewGoBuildTask(fs, projectInfo)
+		isGoBuildable, err := goBuildable(fs, projectInfo.WorkingDirectory)
 		if err != nil {
 			return nil, microerror.Mask(err)
 		}
-		w = append(w, goBuild)
+		if isGoBuildable {
+			goBuild, err := NewGoBuildTask(fs, projectInfo)
```
Comment (JosephSalisbury) [inline workflow/workflow.go]: `go build` on libraries doesn't really make sense.

File: workflow/workflow_test.go [hunk 3]
```diff
@@ -146,7 +167,7 @@ func TestGetBuildWorkflow(t *testing.T) {
 			},
 		},
 
-		// Test 5 a project with golang files, and a dockerfile produces a correct
```
Comment (xh3b4sd) [inline workflow/workflow_test.go]: Please put the numbers back. They are really helpful to me.

File: workflow/workflow_test.go [hunk 4]
```diff
@@ -177,23 +198,23 @@ func TestGetBuildWorkflow(t *testing.T) {
 	for index, test := range tests {
 		fs := afero.NewMemMapFs()
 		if err := test.setUp(fs); err != nil {
-			t.Fatalf("test %d received unexpected error during setup: %#v", index+1, err)
+			t.Fatalf("test %d received unexpected error during setup: %#v", index, err)
```
Comment (kopiczko) [inline workflow/workflow_test.go]: +1 on indexing from 0. And by that I mean I like indexing tests from 0 :D

**Pull request merged**